wagtail/wagtailsearch/backends/elasticsearch2.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import absolute_import, unicode_literals
+
+from wagtail.wagtailsearch.index import FilterField, RelatedFields, SearchField
+
+from .elasticsearch import (
+    ElasticsearchIndex, ElasticsearchMapping, ElasticsearchSearchBackend, ElasticsearchSearchQuery,
+    ElasticsearchSearchResults)
+
+
+def get_model_root(model):
+    """
+    This function finds the root model for any given model. The root model is
+    the highest concrete model that it descends from. If the model doesn't
+    descend from another concrete model then the model is it's own root model so
+    it is returned.
+
+    Examples:
+    >>> get_model_root(wagtailcore.Page)
+    wagtailcore.Page
+
+    >>> get_model_root(myapp.HomePage)
+    wagtailcore.Page
+
+    >>> get_model_root(wagtailimages.Image)
+    wagtailimages.Image
+    """
+    if model._meta.parents:
+        parent_model = list(model._meta.parents.items())[0][0]
```

Comment:
This gives me an error: ``TypeError: 'odict_keys' object does not support indexing``